dashboard/app/models/pd/enrollment.rb

```diff
@@ -226,6 +227,8 @@ def autoupdate_user_field
   end
 
   def enroll_in_corresponding_online_learning
+    autoupdate_user_field if email_changed?
```

Comment:
Note, this is already called [before_validation](https://github.com/code-dot-org/code-dot-org/blob/0d5d2c65a4181b9ab60f14e9038671cef9c51254/dashboard/app/models/pd/enrollment.rb#L60) in all cases, which is called before save as long as validation is not suppressed, so this line shouldn't be needed.  

I think we're good just by switching from `after_create` to `after_save` (with constraints) above. Try running the tests without this.